### PR TITLE
Add DriverDetails::BUG to Disable "LoadOp" clear renderpass in vulkan

### PR DESCRIPTION
**Pull request by JonnyH**
This optimisation doesn't work on PowerVR's Vulkan implementation. We do
not emit tiles (and thus don't do the loadop/storeop) for regions of the
screen where there is no drawn geometry in this renderpass. This causes
a number of issues as there are many scenes where this causes the data
not to be cleared, and end up keeping the stale data.

This fixes the windwaker intro "smearing"

(This PR is for discussion, I don't know what the Vulkan spec says
*should* happen here, if the PVR implementation is "correct" this may
affect multiple vendor's drivers - so don't actually expect it to be merged)

### CODE REVIEW

**Comment by stenzek** [comment]
I only added the clear render pass as a potential optimization for tilers, so if it's not having that affect, then we may as well drop it entirely.

I made the assumption that beginning a render pass with the ops set to clear/store would write the clear value back to the framebuffer. If this is not in line with the spec, another reason to drop it. Although, it does seem strange that drawing say, a triangle that covered half the render area, would result in the other half not being cleared.

File: Source/Core/VideoBackends/Vulkan/Renderer.cpp [hunk 1]
```diff
@@ -378,6 +379,13 @@ void Renderer::ClearScreen(const EFBRectangle& rc, bool color_enable, bool alpha
   // If we're not in a render pass (start of the frame), we can use a clear render pass
   // to discard the data, rather than loading and then clearing.
   bool use_clear_render_pass = (color_enable && alpha_enable && z_enable);
+  if (DriverDetails::HasBug(
+          DriverDetails::BUG_LOADOP_CLEAR_DOESNT_CLEAR_PIXELS_WITH_NO_DRAWN_GEOMETRY))
+  {
+    // This path cannot be used if the driver implementation doesn't guarantee pixels with no drawn
+    // geomerty in "this" renderpass won't be cleared
```
Comment (BhaaLseN) [inline Source/Core/VideoBackends/Vulkan/Renderer.cpp]: geomerty ;)
Might wanna fix this typo if it does end up getting merged.

**Comment by JonnyH** [comment]
I didn't have much time to look at this today, but I can't see *why* we aren't drawing those tiles - reducing this to a simple "empty" renderpass (vkBeginRenderPass();vkEndRenderPass();) doesn't clear it correctly, while it does when pulling that out into a standalone unit test.

Which makes me think this is becoming more likely to be a driver bug than some weird spec thing.

**Comment by JonnyH** [comment]
This is a driver bug, it *should* be fixed on some yet-to-be-released drivers, but that's not useful right now.

The issue is that the framebuffer object effectively ends up keeping some state - namely the load/store op it was created with, and doesn't correctly use the renderpass' version, despite the spec saying they should be 'compatible' and should therefore work.

File: Source/Core/VideoCommon/DriverDetails.cpp [hunk 2]
```diff
@@ -103,7 +103,10 @@ static BugInfo m_known_bugs[] = {
     {API_OPENGL, OS_LINUX, VENDOR_MESA, DRIVER_NOUVEAU, Family::UNKNOWN,
      BUG_SHARED_CONTEXT_SHADER_COMPILATION, -1.0, -1.0, true},
     {API_VULKAN, OS_ALL, VENDOR_NVIDIA, DRIVER_NVIDIA, Family::UNKNOWN,
-     BUG_BROKEN_MSAA_VKCMDCLEARATTACHMENTS, -1.0, -1.0, true}};
+     BUG_BROKEN_MSAA_VKCMDCLEARATTACHMENTS, -1.0, -1.0, true},
+    {API_VULKAN, OS_ALL, VENDOR_IMGTEC, DRIVER_IMGTEC, Family::UNKNOWN,
+     BUG_BROKEN_CLEAR_LOADOP_RENDERPASS, -1.0, -1.0, true},
```
Comment (riking) [inline Source/Core/VideoCommon/DriverDetails.cpp]: If you know the start/end versions, you can fill out the -1.0 fields (end-at version in particular).

Comment (JonnyH) [reply Source/Core/VideoCommon/DriverDetails.cpp]: Currently, the ImgTec driver's version string isn't parsed in OGL, and the vk 'driverVersion' ends up following a completely different versioning system to the software releases.

**Comment by JonnyH** [comment]
I think I'm happy with this to be reviewed and merged, if:
- This is considered an acceptable way of disabling it (IE it still ends up creating the Framebuffer and RenderPass that are then never used)
- this whole thing isn't just removed (e.g. as in PR #5931)
- The performance impact isn't too high that it's worth the extra SW complexity of creating a different Framebuffer object for different loadops (IE the "real" PowerVR driver bug this is working around)

**Pull request merged**